cupy/_core/_kernel.pyx

```diff
@@ -953,6 +932,10 @@ cdef str fix_cast_expr(src_type, dst_type, str expr):
         if dst_kind == 'b':
             return f'({expr}) != {_scalar.get_typename(src_type)}()'
         else:  # dst_kind in 'iuf' (int, uint, float)
+            # give ComplexWarning, as we are explictly using `.real()`
+            warnings.warn(
+                'Casting complex values to real discards the imaginary part',
+                numpy.ComplexWarning)
```

Comment:
I am not sure I like this placement... But, since both the warning and the `.real()` fix are very closely related, it seemed sensible to merge the two.